contracts/earn/strategies/SpookySwap/StrategyLPV2.sol

```diff
@@ -0,0 +1,211 @@
+            uint256 perfAmount = wantAfter.sub(wantBefore);
+            chargePerformanceFee(perfAmount);
+        }
+        deposit();
+    }
+
+    function deposit() public override whenNotPaused nonReentrant {
+        uint256 wantBalance = IERC20(want).balanceOf(address(this));
+        if (wantBalance > 0) {
+            IMasterChefV2(masterChefV2).deposit(pid, wantBalance);
+        }
+    }
+
+    function balanceOfPool() public view override returns (uint256) {
+        (uint256 amount, ) = IMasterChefV2(masterChefV2).userInfo(pid, address(this));
+        return amount;
+    }
+
+    /* ----- Private Functions ----- */
+
+    function _giveAllowances() internal override {
+        TransferHelper.safeApprove(want, masterChefV2, 0);
+        TransferHelper.safeApprove(want, masterChefV2, uint256(-1));
+
+        TransferHelper.safeApprove(reward, uniRouter, 0);
+        TransferHelper.safeApprove(reward, uniRouter, uint256(-1));
+
+        TransferHelper.safeApprove(lpToken0, uniRouter, 0);
+        TransferHelper.safeApprove(lpToken0, uniRouter, uint256(-1));
+
+        TransferHelper.safeApprove(lpToken1, uniRouter, 0);
+        TransferHelper.safeApprove(lpToken1, uniRouter, uint256(-1));
+    }
+
+    function _removeAllowances() internal override {
+        TransferHelper.safeApprove(want, masterChefV2, 0);
+        TransferHelper.safeApprove(reward, uniRouter, 0);
+        TransferHelper.safeApprove(lpToken0, uniRouter, 0);
+        TransferHelper.safeApprove(lpToken1, uniRouter, 0);
+    }
+
+    function _addLiquidity() private {
+        uint256 rewardHalf = IERC20(reward).balanceOf(address(this)).div(2);
+
+        if (lpToken0 != reward) {
+            IPancakeRouter(uniRouter).swapExactTokensForTokens(rewardHalf, 0, rewardToLP0Route, address(this), now);
+        }
+
+        if (lpToken1 != reward) {
+            IPancakeRouter(uniRouter).swapExactTokensForTokens(rewardHalf, 0, rewardToLP1Route, address(this), now);
+        }
+
+        uint256 lp0Balance = IERC20(lpToken0).balanceOf(address(this));
+        uint256 lp1Balance = IERC20(lpToken1).balanceOf(address(this));
+        IPancakeRouter(uniRouter).addLiquidity(lpToken0, lpToken1, lp0Balance, lp1Balance, 0, 0, address(this), now);
+    }
+
+    function retireStrat() external override {
+        require(msg.sender == vault, 'StrategySpookySwapLPV2: NOT_VAULT');
+        IMasterChefV2(masterChefV2).emergencyWithdraw(pid, address(this));
+        uint256 wantBalance = IERC20(want).balanceOf(address(this));
+        if (wantBalance > 0) {
+            TransferHelper.safeTransfer(want, vault, wantBalance);
+        }
+    }
+
+    function emergencyExit() external override onlyAdmin {
+        IMasterChefV2(masterChefV2).emergencyWithdraw(pid, address(this));
+        uint256 wantBalance = IERC20(want).balanceOf(address(this));
+        if (wantBalance > 0) {
+            TransferHelper.safeTransfer(want, vault, wantBalance);
+        }
```

Comment:
What about ? (I'm not sure, just a suggestion)

          IMasterChefV2(masterChefV2).emergencyWithdraw(pid, vault);

Reply:
Same for the method above.

Reply:
It works, good suggestion!